### PR TITLE
Make solc quit if no arguments are supplied to the compiler.

### PR DESCRIPTION
**Pull request by VoR0220**
something that has always bugged me so I took action and decided to fix it :)

### CODE REVIEW

**Comment by VoR0220** [comment]
strange...for some reason Soltest does not like this. Any ideas?

**Comment by chriseth** [comment]
This changes functionality: You can use `solc` to compile stdin.

**Comment by VoR0220** [comment]
I'm sorry, can you elaborate?

**Comment by chriseth** [comment]
You can use `echo 'contract x { function f() { ... } } ' | solc` to compile a contract.

**Comment by VoR0220** [comment]
ahhh...do you think eventually we could add a scripting feature akin to node.js/python? Because having it go blank after a mistyped solc is somewhat irritating.

**Comment by chriseth** [comment]
You mean a repl feature? I don't think so, unless we change the outermost context to allow statements.

**Comment by chriseth** [comment]
Ah, just got an idea:
If add a check for stdin being a terminal, then solc can show the help screen, I suppose.

**Comment by VoR0220** [comment]
@chriseth that would be perfect!

File: solc/main.cpp [hunk 1]
```diff
@@ -29,6 +35,15 @@ using namespace std;
 int main(int argc, char** argv)
 {
 	dev::solidity::CommandLineInterface cli;
+
+	if (isatty(fileno(stdin)) && argc == 1)
```
Comment (chriseth) [inline solc/main.cpp]: I think this should rather go into `CommandLineInterface.cpp`.

File: solc/main.cpp [hunk 2]
```diff
@@ -29,6 +35,15 @@ using namespace std;
...
+	{
+		argc = 2;
+		argv++;
+		argv[0] = (char*)("solc");
+		argv[1] = (char*)("--help");
+		argv[2] = NULL;
```
Comment (chriseth) [inline solc/main.cpp]: There is no guarantee that `argv[2]` is valid.

Comment (VoR0220) [inline solc/main.cpp]: Thats why we check that argc is 1...that way we know argv[2] is invalid and fine to set to null.

File: solc/main.cpp (hunk 1 above)
Comment (VoR0220) [inline solc/main.cpp]: That could work.

**Pull request merged**